### DISCUSSION TITLE
(help) Change data directory on windows and portability

### DISCUSSION THREAD

**Discussion by cptvincer** [TriliumNext/Trilium#1974] [Q&A] [answered]
Sorry for not being tech savvy enough to understand, but the instructions on the wiki are beyond me:
https://github.com/zadam/trilium/wiki/Data-directory
It only mentions how to do it in Linux and MacOs.
Ive installed it for desktop on windows, and i have no idea where is the TRILIUM_DATA_DIR environment variable that i can change for windows

This is the only thing stopping me from using Trilium- im loving it, but i need to add files(a lot) for reference and my appdata on c:/ is on a small ssd with little storage.

Another thing im missing is having separate instances and portability.
-Can i have 2 trillium instances running at the same time tied to different data directories/document.db? (on windows)
-can i install it on an external storage, changing its data directory accordingly?

can anyone help me? Ive been perusing all alternatives out there (i think i went through every alternative available) and none fits my needs as much as trillium, but i cant use in my machine that way

Ps: i would love to see this made easier via the app itself- somewhere in settings a tab for changing/moving data directory...
Even better if the same installation can open/load already existing data directories, that way someone can switch between different libraries for different usecases

Using a single db for vastly differently fields (3 in my case) makes it harder to move or backup then accordingly (i need only one in another machine for example), makes the suggested notes go crazy and useless and adds a lot of noise when searching or making a link (like similar named notes in different subjects)

**Comment by XiaofengdiZhu** [comment, marked as answer]
For setting TRILIUM_DATA_DIR: 
![Snipaste_2021-05-24_20-47-59](https://user-images.githubusercontent.com/16849698/119350049-714d2500-bcd1-11eb-9c39-f28a236b2acb.png)

**Reply by cptvincer** [reply]
So simple! thanks a lot! That screenshot alone added to the wiki would be wonderful.

It must be run through the .bat everytime?

And can that run multiple instances at the same time- like one window running on data_dir, another one using a different one?

**Reply by XiaofengdiZhu** [reply]
Try set TRILIUM_DATA_DIR as a System variable:[https://www.computerhope.com/issues/ch000549.htm](https://www.computerhope.com/issues/ch000549.htm) 
I don't know wheather it will work.
I find a way to run multiple instances: add one line `SET TRILIUM_PORT=37841`(or other number) before `start trilium.exe` in the .bat file.
